application-forum-ui/src/main/resources/ForumCode/Ratings.xml

```diff
@@ -38,12 +38,16 @@
   <hidden>true</hidden>
   <content>{{velocity}}
 #set ($discard = $response.setContentType('application/json'))
-#set ($docRef = $request.docRef)
+#set ($docRef = $services.model.resolveDocument($request.docRef))
 #if (!$isGuest &amp;&amp; "$!{request.vote}" == '1')
-  #set ($discard = $services.ratings.setRating($docRef, $xcontext.user, 1))
+  #set ($discard = $services.ratings.setRating($docRef, $xcontext.userReference, 1))
 #end
 #set ($avgRating = $services.ratings.getAverageRating($docRef))
 #set ($nbVotes = $avgRating.nbVotes)
+## Newer versions of services.ratings API return Optional objects so we have to retrieve the value from within.
+#if ("$!nbVotes" == '')
+  #set ($nbVotes = $avgRating.get().getNbVotes())
+#end
```

Comment:
Same comment as above. You should try the latest API first and then fallback on the old APIs. Plus, don't forget to specify the XWiki version.